src/GovUk.Education.ExploreEducationStatistics.Admin/Services/DataGuidanceMigrationService.cs

```diff
@@ -0,0 +1,83 @@
+﻿#nullable enable
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using GovUk.Education.ExploreEducationStatistics.Admin.Services.Interfaces;
+using GovUk.Education.ExploreEducationStatistics.Admin.Services.Interfaces.Security;
+using GovUk.Education.ExploreEducationStatistics.Admin.ViewModels;
+using GovUk.Education.ExploreEducationStatistics.Common.Model;
+using GovUk.Education.ExploreEducationStatistics.Common.Services.Interfaces.Security;
+using GovUk.Education.ExploreEducationStatistics.Content.Model;
+using GovUk.Education.ExploreEducationStatistics.Content.Model.Database;
+using GovUk.Education.ExploreEducationStatistics.Data.Model.Database;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.EntityFrameworkCore;
+
+namespace GovUk.Education.ExploreEducationStatistics.Admin.Services;
+
+/// <summary>
+/// TODO EES-4661 Remove after the EES-4660 data guidance migration is successful
+/// </summary>
+public class DataGuidanceMigrationService : IDataGuidanceMigrationService
+{
+    private readonly ContentDbContext _contentDbContext;
+    private readonly StatisticsDbContext _statisticsDbContext;
+    private readonly IUserService _userService;
+
+    public DataGuidanceMigrationService(ContentDbContext contentDbContext,
+        StatisticsDbContext statisticsDbContext,
+        IUserService userService)
+    {
+        _contentDbContext = contentDbContext;
+        _statisticsDbContext = statisticsDbContext;
+        _userService = userService;
+    }
+
+    public async Task<Either<ActionResult, DataGuidanceMigrationReport>> MigrateDataGuidance(bool dryRun = true,
```

Comment:
Since this is ultimately going to be removed, I'd have put this all in the controller file, then you'd save yourself having to create the service, the interface, and adding the Startup.cs line. I'd have defined `DataGuidanceMigrationReport` in the controller file too.

Not saying change this now it's done - I think this is a super minor point - but I think it would have saved effort both in creating the migration endpoint and in removing it - and less chance something gets missed when removing it.

Although maybe it would have been more difficult to unit test...

Reply:
I already saw that you did this in `MethodologyMigrationController` but it was no big deal to create the extra classes / interfaces and I've got comments on stuff to help identity what to remove later.

Part of the reason I did it in this way was to roll out our standard pattern of unit testing the service behaviour and adding a permissions test too.